File: 2.Regression/1.Simple Regression/R/simple_linear_regression.R
# Simple Linear Regression

# importing Dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[ ,2:3]

#Splitting the dataset
#install.packages('caTools') this used to install libraries

library(caTools) # to import libraries
split = sample.split(dataset$Salary ,SplitRatio = 2/3) # SplitRatio means ratio for training set 
training_set = subset(dataset , split == TRUE)
test_set = subset(dataset , split == FALSE)

# feature scaling 
# training_set[ ,2:3] = scale(training_set[ ,2:3]) #we choose col 2 and 3 because col 1and 4 aren't numerical values  
# test_set[ ,2:3] = scale(test_set[ ,2:3])

# Fitting simple linear regression to training set
regressor = lm(formula = Salary ~ YearsExperience
               ,data = training_set)

# predicting the test set result
y_pred = predict(regressor, newdata = test_set)

# visualizing the training set result
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience ,y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience ,y = predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary vs Experience') +
  xlab('Years Of Experience') +
  ylab('Salary')

# visualizing the test set result
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience ,y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience ,y = predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary vs Experience') +
  xlab('Years Of Experience') +
  ylab('Salary')

File: 2.Regression/1.Simple Regression/Python/simple_linear_regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Sep 16 23:10:32 2020

@author: amira_mohamed
"""
# Simple Linear Regression

#Importing libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

#Importing dataset
dataset = pd.read_csv('Salary_Data.csv')
#creating matrix with 3 features without last col
X = dataset.iloc[: ,:-1].values
y = dataset.iloc[: ,1].values

# Splitting the dataset 
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X, y, test_size =1/3 ,random_state =0)

# feature scaling 
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.fit_transform(X_test)"""

# Fitting simple linear regression to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

""" 
1. First we import class from the library of each algorithm
2. Create object of this class
3. Fit data By this object
"""

# predicting the test set result
y_pred = regressor.predict(X_test)

# visualizing the training set result
plt.scatter(X_train, y_train, color ='red')
plt.plot(X_train, regressor.predict(X_train),color ='blue')
plt.title('Salary Vs Experience')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()